### PR TITLE
Add a label for the timestamp pointed by the mouse

### PR DESCRIPTION
**Pull request by karupayun**
In this PR we are adding a new label for the timestamp pointed by the
mouse. This label has 2 more digits of precision.

The label can be found in http://screenshot/3DBDt7K8CVgiwUr.

I made a few refactors:
- I simplify kLeftMargin and kRightMargin and kPixelMargin by only using
  kLabelPadding for the 4 dimensions.
- The label has to be drawn in Draw(), since it has to be updated in
  every frame, so we need to save the number of digits of places needed
  for the labels.

http://b/218312292

TODO: Timeline should be below the green line pointed by the mouse, as
well the green overlay (right click). Will come in a following PR.

### CODE REVIEW

**Review by pierricgimmig — COMMENTED** [review]
Hi Pablo, thanks for the new feature! I have one main comment about the mutable member variable and `const` setter.

File: src/OrbitGl/TimelineUi.cpp [hunk 1]
```diff
@@ -138,20 +144,27 @@ std::vector<uint64_t> TimelineUi::GetTicksForNonOverlappingLabels(
...
-                                   uint32_t number_of_decimal_places) const {
+                                   const std::vector<uint64_t>& tick_list) const {
   if (tick_list.size() <= 1) return false;
   float distance_between_labels = GetTickWorldXPos(tick_list[1]) - GetTickWorldXPos(tick_list[0]);
   for (auto tick_ns : tick_list) {
     float label_width = text_renderer.GetStringWidth(
-        GetLabel(tick_ns, number_of_decimal_places).c_str(), layout_->GetFontSize());
-    if (distance_between_labels < horizontal_margin + label_width) {
+        GetLabel(tick_ns, GetNumberOfDecimalPlacesNeeded()).c_str(), layout_->GetFontSize());
+    if (distance_between_labels < 2 * kLabelsPadding + label_width) {
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/TimelineUi.cpp]: I'm forgetting all this 2.f, thanks for catching them!

File: src/OrbitGl/TimelineUi.h [hunk 2]
```diff
@@ -33,23 +35,27 @@ class TimelineUi : public CaptureViewElement {
...
+  [[nodiscard]] uint32_t GetNumberOfDecimalPlacesNeeded() const {
+    return number_of_decimal_places_needed_;
+  }
+  void SetNumberOfDecimalPlacesNeeded(uint32_t number_of_decimal_places_needed) const {
+    number_of_decimal_places_needed_ = number_of_decimal_places_needed;
+  }
 
   const TimelineInfoInterface* timeline_info_interface_;
   TimelineTicks timeline_ticks_;
+  mutable uint32_t number_of_decimal_places_needed_ = 0;
```
[the remark this answers is not in the document]
Comment (karupayun) [reply src/OrbitGl/TimelineUi.h]: Yes, I also didn't like it but I thought that was the best solution. I've just refactored, let me know what you think about now.

**Review by pierricgimmig — APPROVED** [review]
Thanks for the change, lgtm!

**Review by beckerhe — APPROVED** [review]
LGTM

**Pull request merged**